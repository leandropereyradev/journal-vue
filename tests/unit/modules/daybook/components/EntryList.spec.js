import { shallowMount } from "@vue/test-utils";

import EntryList from "@/modules/daybook/components/EntryList.vue";

import { journalState } from "../../../mocks/test-journal-state";
import createVuexStore from "../../../mocks/createVuexStore";
import mockRouter from "../../..//mocks/mockRouter";

describe("EntryList component", () => {
  const store = createVuexStore(journalState);

  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();

    wrapper = shallowMount(EntryList, {
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });
  });

  test("Should call getEntriesByTerm without term and show 2 entries and match with snapshot", () => {
    expect(wrapper.findAll("entry-stub").length).toBe(2);

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Should call the getEntriesByTerm and filter the entries", async () => {
    const input = wrapper.find("input");
    await input.setValue("segunda");

    expect(wrapper.findAll("entry-stub").length).toBe(1);
  });

  test("The 'New' button should redirect to /new", () => {
    wrapper.find("button").trigger("click");

    expect(mockRouter.push).toHaveBeenCalledWith({
      name: "entry",
      params: { id: "new" },
    });
  });
});
