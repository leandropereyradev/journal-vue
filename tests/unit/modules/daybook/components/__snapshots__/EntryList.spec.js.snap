// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryList component Should call getEntriesByTerm without term and show 2 entries and match with snapshot 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="text" class="form-control" placeholder="Buscar entrada"></div>
  <div class="mt-2 d-flex flex-column"><button class="btn btn-primary mx-3"><i class="fa fa-plus-circle"></i> Nueva Entrada </button></div>
  <div class="entry-scrollarea">
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
  </div>
</div>
`;
